receivers/alertmanager/config.go

```diff
@@ -17,13 +16,13 @@ type Config struct {
 	Password string
 }
 
-func ValidateConfig(fc receivers.FactoryConfig) (Config, error) {
+func ValidateConfig(jsonData json.RawMessage, decryptFn receivers.DecryptFunc) (Config, error) {
```

Comment:
Now that I fully have the context of this change - I'd say let's do one more refactor here and rename this method on all receivers from `ValidateConfig` to `NewConfig`.

`ValidateConfig` is wrong and deceiving, when you take some primitives and return a new struct this is way more than a validation.